File: iris-crypto/iris-crypto-rsa/src/main/java/city/smartb/iris/crypto/rsa/signer/RS256Signer.java
package city.smartb.iris.crypto.rsa.signer;

import com.nimbusds.jose.JWSAlgorithm;

import java.security.GeneralSecurityException;
import java.security.Signature;
import java.security.interfaces.RSAPrivateKey;

public class RS256Signer implements Signer {

	private final RSAPrivateKey rsaPrivateKey;

	public RS256Signer(RSAPrivateKey rsaPrivateKey) {
		this.rsaPrivateKey = rsaPrivateKey;
	}

	@Override
	public JWSAlgorithm getAlgorithm() {
		return  JWSAlgorithm.RS256;
	}

	@Override
	public String getTerm() {
		return "RsaSignature2018";
	}

	@Override
	public byte[] sign(byte[] content) throws GeneralSecurityException {
		Signature jcaSignature = Signature.getInstance("SHA256withRSA");
		jcaSignature.initSign(rsaPrivateKey);
		jcaSignature.update(content);
		return jcaSignature.sign();
	}
}

File: iris-crypto/iris-crypto-rsa-key/src/main/kotlin/city/smartb/iris/crypto/rsa/signer/RS256Signer.kt
package city.smartb.iris.crypto.rsa.signer

import city.smartb.iris.crypto.dsl.signer.Signer
import com.nimbusds.jose.JWSAlgorithm
import java.security.GeneralSecurityException
import java.security.Signature
import java.security.interfaces.RSAPrivateKey

class RS256Signer(
    private val rsaPrivateKey: RSAPrivateKey
) : Signer {

    override val algorithm: JWSAlgorithm
        get() = JWSAlgorithm.RS256
    override val term: String
        get() = "RsaSignature2018"

    @Throws(GeneralSecurityException::class)
    override fun sign(content: ByteArray): ByteArray {
        val jcaSignature = Signature.getInstance("SHA256withRSA")
        jcaSignature.initSign(rsaPrivateKey)
        jcaSignature.update(content)
        return jcaSignature.sign()
    }
}
